src/test/scala/com/databricks/spark/redshift/RedshiftSuite.scala

```diff
@@ -0,0 +1,458 @@
+
+    conn.prepareStatement("drop table if exists test_table").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table2").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table3").executeUpdate()
+    conn.commit()
+
+    def createTable(tableName: String): Unit = {
+      conn.prepareStatement(
+        s"""
+           |create table $tableName (
+           |testbyte int2,
+           |testbool boolean,
+           |testdate date,
+           |testdouble float8,
+           |testfloat float4,
+           |testint int4,
+           |testlong int8,
+           |testshort int2,
+           |teststring varchar(256),
+           |testtimestamp timestamp
+           |)
+      """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |null, null, null, null, null, null, null, null, null, null)
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |0, null, '2015-07-03', 0.0, -1.0, 4141214, 1239012341823719, null, 'f', '2015-07-03 00:00:00.000')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |0, false, null, -1234152.12312498, 100000.0, null, 1239012341823719, 24, '___|_123', null)
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |1, false, '2015-07-02', 0.0, 0.0, 42, 1239012341823719, -13, 'asdf', '2015-07-02 00:00:00.000')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |1, true, '2015-07-01', 1234152.12312498, 1.0, 42,
+           |1239012341823719, 23, 'Unicode''s樂趣', '2015-07-01 00:00:00.001')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.commit()
+    }
+
+    createTable("test_table")
+    createTable("test_table2")
+    createTable("test_table3")
+
+    conn.commit()
+  }
+
+  override def afterAll(): Unit = {
+    val temp = new File(tempDir)
+    val tempFiles = temp.listFiles()
+    if(tempFiles != null) tempFiles foreach {
+      case f => if(f != null) f.delete()
+    }
+    temp.delete()
+
+    conn.prepareStatement("drop table if exists test_table").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table2").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table3").executeUpdate()
+    conn.commit()
+    conn.close()
+
+    sc.stop()
+    super.afterAll()
+  }
+  
+  test("DefaultSource can load Redshift UNLOAD output to a DataFrame") {
+    val sqlContext = new SQLContext(sc)
+    sqlContext.sql(
+      s"""
+         |create temporary table test_table(
```

Comment:
It looks like every test case creates its own temporary table on a fresh SQL context.  I think that it would make more sense to extract this out into a `beforeEach` method, since the code duplication here makes the tests fairly hard to read.